ledger/src/blockstore_processor.rs

```diff
@@ -214,6 +218,38 @@ pub fn execute_batch(
     first_err.map(|(result, _)| result).unwrap_or(Ok(()))
 }
 
+#[derive(Debug, Default)]
+pub struct ThreadExecuteTimings {
+    pub total_thread_us: u64,
+    pub total_transactions_executed: u64,
+    pub execute_timings: ExecuteTimings,
+}
+
+impl ThreadExecuteTimings {
+    pub fn report_stats(&self, slot: Slot) {
+        lazy! {
+            datapoint_info!(
+                "replay-slot-end-to-end-stats",
```

Comment:
I don't want to punish a good deed 😉 , but it has annoyed me that these datapoints (this one and the one below) both mention `replay` specifically, but neither live with replay code:
https://github.com/anza-xyz/agave/blob/70c4cb0ba1fbef2b3aa5796420ed285c46f5c4f6/core/src/consensus/progress_map.rs#L49-L50

Granted the one I just linked is `ReplaySlotStats` which is technically `ReplayStage` specific, but `ReplaySlotStats` is an aliased `ConfirmationTiming` (which is in `blockstore_processor`) with a few extra values that get passed in from `ReplayStage` main loop.

In any case, the one I mentioned invokes the one you're moving here:
https://github.com/anza-xyz/agave/blob/70c4cb0ba1fbef2b3aa5796420ed285c46f5c4f6/core/src/consensus/progress_map.rs#L86

I think it'd be nice to have these next to each other in code, potentially both in `blockstore_processor`. Would you be up for moving this other one I mentioned as well ?